CMakeLists.txt

```diff
@@ -105,18 +105,20 @@ if(${BUILD_UNSTABLE_TOOLS})
 	add_subdirectory(Tools/GeneratorPerformanceTest/)
 endif()
 
-include(CheckCXXCompilerFlag)
+# The Android toolchain sets its own flags; interference is bad
 include(SetFlags.cmake)
-set_flags()
-set_lib_flags()
-enable_profile()
+if (NOT ANDROID)
```

Comment:
I'm still not happy with this. There is no way the android toolchain can inject different flags for libraries, and executables, which means this will break things.